src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6600,7 +6605,7 @@ private bool IsPossibleLocalDeclarationStatement(bool allowAnyExpression)
             }
 
             tk = this.CurrentToken.ContextualKind;
-            if (IsAdditionalLocalFunctionModifier(tk) &&
+            if (IsAdditionalLocalFunctionModifier(tk) || tk == SyntaxKind.OpenBracketToken &&
                 (tk != SyntaxKind.AsyncKeyword || ShouldAsyncBeTreatedAsModifier(parsingStatementNotDeclaration: true)))
```

Comment:
i find this really hard to parse.  consider more parens... or extracting into helper function (where these are doc'ed statements), or making locals with comments on them.  

--

specifically, i could not answer why you made this change (i.e. why the openbracket change).  On top of not understanding that, the code also feels actively confusing, making it more difficult to understand why you'd make that change.  Once code gets to that point, simplification and documentation are super helpful.  Thanks!

Reply:
(oh, i'm also an idiot since i read this as `Open Brace` not `Open Bracket`.  But the above points still stand).

Reply:
I think extracting part of the condition into a named local (something like `isPossibleLocalFunction`?) could be helpful.